src/sqlparser/src/ast/mod.rs

```diff
@@ -106,19 +110,57 @@ impl Ident {
         }
     }
 
-    /// Create a new quoted identifier with the given quote and value. This function
-    /// panics if the given quote is not a valid quote character.
-    pub fn with_quote<S>(quote: char, value: S) -> Self
+    /// Create a new identifier from word
+    pub fn new_from_word(w: &Word) -> Result<Ident, ParserError> {
+        if w.value.is_empty() {
+            Err(ParserError::ParserError(format!(
+                "zero-length delimited identifier at or near \"{w}\""
+            )))
+        } else {
+            Ok(Ident {
+                value: w.value.clone(),
+                quote_style: w.quote_style,
+            })
+        }
+    }
+
+    /// Create a new quoted identifier with the given quote and value.
+    /// the inputs are guanteed to be legal.
+    pub fn new_with_quote_safe<S>(quote: char, value: S) -> Self
```

Comment:
```suggestion
    pub fn with_quote_unchecked<S>(quote: char, value: S) -> Self
```

According to rust naming convention. https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case:~:text=new%20or-,with_more_details,-Conversion%20constructors